File: client/Assets/Scripts/bookrpg/data/DataItemBase.cs
///
/// Copyright (c) 2016, bookrpg, All rights reserved.
/// @author llj <wwwllj1985@163.com>
/// @license The MIT License
///

using System;
using System.Collections;

namespace bookrpg.data
{
    public abstract class DataItemBase
    {
        protected object _key1;
        protected object _key2;

        public DataItemBase()
        {
        }

        public virtual bool ParseFrom(IDataParser parser)
        {
            throw new NotImplementedException();
        }

        public virtual object GetKey()
        {
            return _key1;
        }

        public virtual object GetSecondKey()
        {
            return _key2;
        }
    }
}

File: server/bookrpg/data/DataItemBase.php
<?php
/**
 * Copyright (c) 2016, bookrpg, All rights reserved.
 * @author llj <wwwllj1985@163.com>
 * @license The MIT License
 */

include_once  __DIR__ . '/IDataParser.php';

class DataItemBase
{
    protected $_key1;
    protected $_key2;

    public function parseFrom(IDataParser $parser)
    {
        throw new Exception('not implements', 1);
    }

    public function getKey()
    {
        return $this->_key1;
    }

    public function getSecondKey()
    {
        return $this->_key2;
    }
}